careers/software-engineer.md

```diff
@@ -57,3 +101,17 @@ An engineer III is capable of taking on complex technical problems and coordinat
 **Communication:** They point out novel opportunities for refactoring during code review. They proactively work to minimize or mitigate potential problems before they arise.
 
 **Collaboration:** Others on the team reach out for help in their area of expertise, and they make themselves available. They are capable of effectively running small to medium-sized projects with little to no coaching, but need help to successfully manage large projects.
+
+### Responsibilities
+
+* Lead the design and development of medium to large-scale software features or systems.
+* Establish and contribute to team-level software architecture and development standards.
+* Mentor and provide technical guidance to junior and mid-level engineers.
+* Collaborate with cross-functional teams to define requirements and contribute to technical roadmaps.
+* Drive the adoption of best practices around coding, design, quality, and performance within your team.
+* Troubleshoot and debug complex technical issues, often involving multiple systems.
+* Participate in code reviews, ensuring that the code we deliver is of the highest possible quality.
+* Participate in and potentially lead technical design discussions and architecture reviews.
+* Understand and contribute to broader business objectives and align technical solutions.
```

Comment:
This reads a bit vague. The "Impact" section above has this bit that feels related, though:

> They can identify strategic opportunities and risks with a proposed plan and affect the direction of the team.

Could we try re-writing this bullet to something like:

> Identify strategic technical opportunities and risks that affect business objectives, and proactively adjust technical direction accordingly.